src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -61,7 +61,7 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
                 unchecked
                 {
                     int unaligned = (int)(byte*)Unsafe.AsPointer(ref searchSpace) & (Vector<byte>.Count - 1);
-                    nLength = (IntPtr)(uint)unaligned;
+                    nLength = (IntPtr)(uint)((Vector<byte>.Count - unaligned) % Vector<byte>.Count);
```

Comment:
Does the `idiv` get elided out in the asm? (otherwise its `&` the jit const `(Vector<byte>.Count - 1)`  on the line above)